tool/test/test_format_ignores.py

```diff
@@ -0,0 +1,223 @@
+"""Test the format_ignores module."""
+
+# ruff: noqa: PLC2701
+import errno
+import sys
+from pathlib import Path
+
+import pytest
+
+from tool.format_ignores import (
+    _process_directory,
+    _process_file,
+    _sort_ignore_list,
+    main,
+)
+
+
+@pytest.mark.parametrize(
+    ("input_text", "expected_output"),
+    [
+        ("error1, error2", "error1, error2"),
+        ("error2, error1", "error1, error2"),
+        ("error1,error2", "error1, error2"),
+        ("error1,  error2", "error1, error2"),
+        ("error1, error1, error2", "error1, error2"),
+        ("", ""),
+        (" , error1, ,error2, ", "error1, error2"),
+    ],
+)
+def test_sort_ignore_list(input_text: str, expected_output: str) -> None:
+    """Test _sort_ignore_list function."""
+    result = _sort_ignore_list(input_text)
+    assert result == expected_output
+
+
+def test_process_file(tmpdir: pytest.TempdirFactory) -> None:
+    """Test _process_file function."""
+    test_file = Path(str(tmpdir)) / "test.pyi"
+    test_content = """def func1() -> None:
+    # type: ignore[attr-defined, no-any-return]
+    pass
+
+    def func2() -> None:
+        # pyright: ignore[reportUnknownMemberType,reportUnknownVariableType]
+        pass
+    """
+    test_file.write_text(test_content, encoding="utf-8")
```

Comment:
Oh haha of course; it doesn't need to be valid python, so indentation doesn't even matter 👌🏻